go/worker/compute/executor/committee/node.go

```diff
@@ -508,6 +509,15 @@ func (n *Node) HandleNewBlockLocked(blk *block.Block) {
 
 		// Record time taken for successfully processing a batch.
 		batchProcessingTime.With(n.getMetricLabels()).Observe(time.Since(state.batchStartTime).Seconds())
+	case StateWaitingForTimeout:
+		// A new block means the round has timeouted.
+		n.logger.Info("considering the round timeouted",
+			"round", blk.Header.Round,
+			"header_hash", blk.Header.EncodedHash(),
+			"header_type", blk.Header.HeaderType,
+		)
+
+		n.transitionLocked(StateWaitingForBatch{})
 	}
```

Comment:
good idea, added it (but it will only schedule immediately if it already has a full batch ready, otherwise it will still wait for the next "force" flush, which i think makes sense)